src/common/mailFunctionality/MailModel.ts

```diff
@@ -354,23 +343,25 @@ export class MailModel {
 					await this._init()
 					m.redraw()
 				}
-			} else if (isUpdateForTypeRef(MailTypeRef, update) && update.operation === OperationType.CREATE) {
-				const folder = this.getMailFolder(update.instanceListId)
-
-				if (folder && folder.folderType === MailFolderType.INBOX && !containsEventOfType(updates, OperationType.DELETE, update.instanceId)) {
-					// If we don't find another delete operation on this email in the batch, then it should be a create operation,
-					// otherwise it's a move
-					const mailId: IdTuple = [update.instanceListId, update.instanceId]
-					const mail = await this.entityClient.load(MailTypeRef, mailId)
-					await this.getMailboxDetailsForMailListId(update.instanceListId)
-						.then((mailboxDetail) => {
-							// We only apply rules on server if we are the leader in case of incoming messages
-							return mailboxDetail && this.inboxRuleHandler.findAndApplyMatchingRule(mailboxDetail, mail, this.connectivityModel.isLeader())
-						})
-						.then((newId) => this._showNotification(newId || mailId))
-						.catch(noOp)
-				}
 			}
+			// FIXME: inbox rules are also applied in MailViewModel, can we get away with just doing it there?
```

Comment:
I think this is resolved later but I'm gonna leave a comment to not lose it

Reply:
it is